beater/beater_test.go

```diff
@@ -0,0 +1,162 @@
+package beater
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	// make sure processors are loaded
+	_ "github.com/elastic/apm-server/processor/error"
+	_ "github.com/elastic/apm-server/processor/transaction"
+
+	"github.com/elastic/apm-server/tests"
+	"github.com/elastic/beats/libbeat/monitoring"
+	"github.com/elastic/beats/libbeat/outputs"
+	pubs "github.com/elastic/beats/libbeat/publisher"
+	"github.com/elastic/beats/libbeat/publisher/pipeline"
+	"github.com/elastic/beats/libbeat/publisher/queue"
+	"github.com/elastic/beats/libbeat/publisher/queue/memqueue"
+)
+
+/*
+Run the benchmarks as follows:
+
+    $ go test beater/*.go -run=XXX -bench=. -cpuprofile=cpu.out -args -e
+ -d *
```

Comment:
Running it locally I just realised this probably felt on the new line by 80 chars magic or soemthing similar. Same below.